ADVANCED.md

```diff
@@ -1151,6 +1152,7 @@ For example, if you add a `search_fields` configuration option, it will control
 | `conditionalFacets`                | Object[String, function] | optional  |         | This facet will only be fetched if the condition specified returns `true`, based on the currently applied filters. This is useful for creating hierarchical facets.<br/><br/>Example: don't return `states` facet data unless `parks` is a selected filter.<br/> `{ states: filters => isParkSelected(filters) }`                                                                                                              |
 | `search_fields`                    | Object[String, Object]   | optional  |         | Fields which should be searched with search term.<br/><br/>[App Search search_fields API Reference](https://swiftype.com/documentation/app-search/api/search/search-fields)                                                                                                                                                                                                                                                    |
 | `result_fields`                    | Object[String, Object]   | optional  |         | Fields which should be returned in results.<br/><br/>[App Search result_fields API Reference](https://swiftype.com/documentation/app-search/api/search/result-fields)                                                                                                                                                                                                                                                          |
+| `spelling`                    | Object  | optional  |         | **Only supported by Site Search** <br/> [Site Search API reference](https://swiftype.com/documentation/site-search/searching/spelling) <br/> Specify the type of spelling behaviour with a `type` field that should have a value of "strict", "always" or "retry" <br/> Example: ```spelling: { type: "strict" }``` |
```

Comment:
Seeing this spelled out in the docs is making me second guess the approach here. It's a bit awkward to have to specify that it is a Site Search only API. I'm also worried about cementing this as part of the API.

I wonder if we could go back to your original approach:
```
    beforeSearchCall: (queryOptions, next)=> {
      const withSpelling = {...queryOptions, spelling: 'strict'}
      return next(withSpelling)
    }
```

Perhaps it would be interesting to change our connectors to store the original, unparsed response in state.

```
state.rawResponse
```

In that case, the AppSearch and SiteSearch connectors would just need to be updated to populate that state field in the response adapters.

It's kind of a fallback; like if all else fails, you can always access the raw response in state. This would solve your case

Another option would be open up a "responseMeta" field in state, which is available for writing arbitrary data to for Connectors. The SiteSearchConnector, in this case, could write out spelling corrections there.

```
state.responseMeta
```

Reply:
Hi, Thanks for the feedback. I agree that this current approach raises some questions, I found myself wondering if I should be adding UI component support, but then realising that would not work with an app search API...

For my purposes at least, having the raw response available would solve the problem so I'd be happy to close this PR and implement that instead

Reply:
I'm going to close this PR for now, I would be happy to see a new PR from you.